src/generate/interrupt.rs

```diff
@@ -180,9 +180,9 @@ pub fn render(
                 #variants
             }
 
-            unsafe impl bare_metal::Nr for Interrupt {
+            unsafe impl cortex_m::interrupt::InterruptNumber for Interrupt {
```

Comment:
This change breaks the build for [`avr-device`](https://github.com/Rahix/avr-device).  The problem is that you modified the `else` branch which is for everything but `msp430` ...

I suppose this code needs to be refactored such that this line is only emitted for `cortex-m`.  I would assume other platforms (like `riscv`) will also run into the same issue.

Cc: @trembel